src/pages/index.tsx

```diff
@@ -3,16 +3,27 @@ import Head from 'next/head'
 
 import Header from 'components/header'
 import Quiz from 'components/quiz'
-export default function Home() {
+import {
+  useAuthUser,
+  withAuthUser,
+  withAuthUserTokenSSR,
+} from 'next-firebase-auth'
+function Home() {
+  const AuthUser = useAuthUser()
+  console.log({AuthUser})
   return (
     <div className="dark:bg-gray-800 bg-gray-50">
       <Head>
         <title>shoulditrain.today</title>
         <link rel="icon" href="/favicon.ico" />
       </Head>
 
-      <Header />
+      <Header email={AuthUser.email} signOut={AuthUser.signOut} />
       <Quiz></Quiz>
     </div>
   )
 }
+
+export const getServerSideProps = withAuthUserTokenSSR()()
```

Comment:
this is how you can use the authed user in `getServerSideProps`. (the whole point of next-firebase-auth)